### ISSUE TITLE
IDE0059 incorrectly marks variables as unnecessary

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#39800]
**Version Used**: 16.4 Preview 4

**Steps to Reproduce**:

```csharp
public static int ToInt32(this string value)
{
    int result = 0;
    _ = int.TryParse(value, out result);
    return result;
}
```

**Expected Behavior**:

The `= 0` is faded as unnecessary.

**Actual Behavior**:

The `result` identifier is faded as unnecessary.

**Comment by sharwell** [comment]
Duplicate of #36202
